java/xtreemfs-servers/src/main/java/org/xtreemfs/mrc/osdselection/FileNamePrefixPolicy.java

```diff
@@ -0,0 +1,97 @@
+package org.xtreemfs.mrc.osdselection;
+
+import org.xtreemfs.common.uuids.ServiceUUID;
+import org.xtreemfs.mrc.metadata.XLocList;
+import org.xtreemfs.pbrpc.generatedinterfaces.DIR;
+import org.xtreemfs.pbrpc.generatedinterfaces.GlobalTypes;
+
+import java.net.InetAddress;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * OSDSelectionPolicy that allows to choose an OSD based on the path of the
+ * currently handled file.
+ */
+public class FileNamePrefixPolicy implements OSDSelectionPolicy {
+
+    public static final short POLICY_ID = (short) GlobalTypes
+            .OSDSelectionPolicyType
+            .OSD_SELECTION_POLICY_FILENAME_PREFIX
+            .getNumber();
+
+    /*
+     identifier for setting prefixes and OSD ids
+    */
+    private static final String attributeKeyString = "filenamePrefix";
+
+    /*
+      prefixes with an explicit OSD
+     */
+    private HashMap<String, String> prefixToOSDMap =
+            new HashMap<String, String>();
+
+    @Override
+    public DIR.ServiceSet.Builder getOSDs(DIR.ServiceSet.Builder allOSDs,
+                                          InetAddress clientIP,
+                                          GlobalTypes.VivaldiCoordinates
+                                                  clientCoords,
+                                          XLocList currentXLoc,
+                                          int numOSDs,
+                                          String path) {
+
+        List<DIR.Service> allServices = allOSDs.getServicesList();
+        List<DIR.Service> returnList = new LinkedList<DIR.Service>();
+
+        if (!this.prefixToOSDMap.isEmpty()) {
+            int prefixLength = path.lastIndexOf("/");
+            String filePrefix = path.substring(0, prefixLength);
```

Comment:
With the correct separator, it will. The path always starts with the volume name and ends with the file name, so there has to be a separator in between.